src/keydra/providers/github.py

```diff
@@ -0,0 +1,238 @@
+        valid, msg = BaseProvider.validate_spec(spec)
+
+        if not valid:
+            return valid, msg
+
+        if 'scope' not in spec:
+            return False, 'Attribute "scope" not present in configuration'
+
+        if 'config' not in spec:
+            return False, 'Attribute "config" not present in configuration'
+
+        if spec['scope'] != 'repository':
+            return False, 'Unsupported scope'
+        else:
+            return Client._validate_repository_spec(spec)
+
+        return True, 'It is valid!'  # pragma: no cover
+
+    @classmethod
+    def pre_process_spec(self, spec, context={}):
+        specs = []
+
+        target = copy.deepcopy(spec)
+
+        current_env_name, current_env = list(
+            context.get('environment', {}).items()
+        )[0]
+
+        if 'ENV' in target['key']:
+            target['key'] = target['key'].format(
+                **{'ENV': current_env_name.upper()}
+            )
+
+        if 'ENV' in target.get('config', {}).get(
+            'environment', ''
+        ):
+            target['config']['environment'] = target[
+                'config'
+            ]['environment'].format(
+                **{'ENV': current_env_name.upper()}
+            )
+
+        if spec['scope'] in ['deployment', 'repository'] and any(
+            [
+                isinstance(spec['config']['repository'], list),
+                isinstance(spec['config']['repository'], tuple)
+            ]
+        ):
+            for repo in spec['config']['repository']:
+                new_target = copy.deepcopy(target)
+                new_target['config']['repository'] = repo
+
+                specs.append(new_target)
+        else:
+            specs.append(target)
+
+        return specs
+
+    def _load_remote_file(self, repo, path, username, filetype):
+        LOGGER.info(
+            'Loading remote file: {} - {}'.format(repo, path)
+        )
+
+        resp = self._client.fetch_file_from_repository(
+            repo=repo,
+            path=path,
+            org=username
+        )
+
+        LOGGER.debug('Remove file content: \n{}'.format(resp))
+
+        if filetype.lower() in ['json']:
+            return json.loads(resp)
+
+        elif filetype.lower() in ['yaml', 'yml']:
+            return yaml.safe_load(resp)
+
+        else:
+            raise ConfigException(
+                'Unsupported filetype provided: {}'.format(filetype)
+            )
+
+    def load_config(self, config):
+        username = config.get('accountusername')
+        secrets_repo = config.get('secrets', {}).get('repository')
+        secrets_path = config.get('secrets', {}).get('path')
+        secrets_filetype = config.get('secrets', {}).get('filetype', 'yaml')
+        envs_repo = config.get('environments', {}).get('repository')
+        envs_path = config.get('environments', {}).get('path')
+        envs_filetype = config.get('environments', {}).get('filetype', 'yaml')
```

Comment:
Might consider moving these strings to consts used by all config providers -- Bitbucket and Github

Reply:
Done